### ISSUE TITLE
Inserting table row throws "Invalid shared formula" exception

### ISSUE THREAD

**Issue by irvalchev** [EPPlusSoftware/EPPlus#594]
I have been working on some workbook with extensive use of formulas. The workbook itself is populated with EPPlus. Once I tried to insert rows in one of the tables, I get the following exception:
```
System.Exception
  HResult=0x80131500
  Message=Invalid shared formula
  Source=EPPlus
  StackTrace:
   at OfficeOpenXml.Core.Worksheet.WorksheetRangeInsertHelper.AdjustFormulasInsert(ExcelRangeBase range, ExcelAddressBase effectedAddress, eShiftTypeInsert shift)
   at OfficeOpenXml.Core.Worksheet.WorksheetRangeInsert
Helper.Insert(ExcelRangeBase range, eShiftTypeInsert shift, Boolean styleCopy)
   at OfficeOpenXml.Table.ExcelTable.InsertRow(Int32 position, Int32 rows, Boolean copyStyles)
```
   
  I think that Excel is somehow updating the tables, since the first time I open them after being populated with EPPlus it asks me to save changes, although none were made.
  
 Here is a code to reproduce the issue:
```
            string path = @"C:\TST.xlsx";
            FileInfo fi = new FileInfo(path);
            using (var package = new ExcelPackage(fi))
            {
                var table = package.Workbook.Worksheets
                    .SelectMany(x => x.Tables)
                    .SingleOrDefault(x => x.Name == "TblComponents");
                table.InsertRow(2, 1, true);
            }
```
 
Also the file I am testing with
[TST.xlsx](https://github.com/EPPlusSoftware/EPPlus/files/8053445/TST.xlsx)

You will notice that some of the named ranges have REF error. This is not the problem, since the same happens if there are no errors. I have managed to pin down the issue to the DU column (the last hidden column). If I remove it and add it again without specifying CalculatedColumnFormula it seems to work. Also when I set the CalculatedColumnFormula to the desired formula I got an argument exception "Invalid address format FIELDCOMPONENTSDU". This is a named range, so I assume there is an issue with the names support.

There is one more issue. I am mentioning it here because it can be reproduced with the same example. When I change the formula of the DU cells to "SUM((IFERROR(TblFmeda[DU],0) & 0) * (TblComponents[[#This Row],[Component]]=TblFmeda[Component Final]))" in the Excel in the end I get "=SUM((IFERROR(TblFmeda[@DU];0)&0)*([@Component]=TblFmeda[@[Component Final]]))", which is incorrect (notice the @ signs).

I have also noticed that when I use table.AddRow, the conditional formattings and cell validations are not applied, that's why I normally use InsertRow. I guess this is a separate issue...

I am using EPPlus version 5.6.0. I've tried it with 5.8.6 with the same success

**Comment by JanKallman** [comment]
Thanks for reporting this. I have applied a fix for the insert issue and the AddRow not extending conditional formatting and data validations. 
The formula is a dynamic arrayformula. That is currently not supported in EPPlus. We will look into dynamic array formulas in coming versions.
You can test this fix by using our latest develop build ... https://github.com/EPPlusSoftware/EPPlus/wiki/Using-our-Develop-Nuget-Feed

**Comment by JanKallman** [comment]
This is fixed in version 5.8.7
